src/Compilers/Core/Portable/PEWriter/MetadataWriter.PortablePdb.cs

```diff
@@ -831,5 +834,128 @@ private void EmbedSourceLink(Stream stream)
                 kind: _debugMetadataOpt.GetOrAddGuid(PortableCustomDebugInfoKinds.SourceLink),
                 value: _debugMetadataOpt.GetOrAddBlob(bytes));
         }
+
+        /// <summary>
+        /// Capture the set of compilation options to allow a compilation 
+        /// to be reconstructed from the pdb
+        /// </summary>
+        private void EmbedCompilationOptions(CommonPEModuleBuilder module)
+        {
+
+            var builder = new BlobBuilder();
+
+            var compilerVersion = typeof(Compilation).Assembly.GetCustomAttribute<AssemblyInformationalVersionAttribute>().InformationalVersion;
+            WriteValue("compilerversion", compilerVersion);
+
+            if (module.EmitOptions.FallbackSourceFileEncoding != null)
+            {
+                WriteValue("fallbackencoding", module.EmitOptions.FallbackSourceFileEncoding.CodePage.ToString());
+            }
+
+            if (module.EmitOptions.DefaultSourceFileEncoding != null)
+            {
+                WriteValue("defaultencoding", module.EmitOptions.DefaultSourceFileEncoding.CodePage.ToString());
+            }
+
+            int portabilityPolicy = 0;
+            if (module.CommonCompilation.Options.AssemblyIdentityComparer is DesktopAssemblyIdentityComparer identityComparer)
+            {
+                portabilityPolicy |= identityComparer.PortabilityPolicy.SuppressSilverlightLibraryAssembliesPortability ? 0b1 : 0;
+                portabilityPolicy |= identityComparer.PortabilityPolicy.SuppressSilverlightPlatformAssembliesPortability ? 0b10 : 0;
+            }
+
+            WriteValue("portability-policy", portabilityPolicy.ToString());
```

Comment:
>- [](start = 35, length = 1)

Should be consistent in using word separator (either we do it for all keys, or for none).